src/backend/commands/createas.c

```diff
@@ -1846,7 +1846,7 @@ create_dynamic_table_auto_refresh_task(ParseState *pstate, Relation DynamicTable
 	dtname = quote_qualified_identifier(get_namespace_name(RelationGetNamespace(DynamicTableRel)),
 											 RelationGetRelationName(DynamicTableRel));
 	resetStringInfo(&buf);
-	appendStringInfo(&buf, "REFRESH DYNAMIC TABLE %s", dtname);
+	appendStringInfo(&buf, "REFRESH DYNAMIC TABLE %s CONCURRENTLY", dtname);
```

Comment:
normal matview support CONCURRENTLY ?

Reply:
Yes, normal materialized views do support the CONCURRENTLY option. According to the [PostgreSQL docs](https://www.postgresql.org/docs/14/sql-refreshmaterializedview.html):

> CONCURRENTLY allows the materialized view to be refreshed without locking out concurrent selects. While a standard refresh might block other connections reading from the view, using CONCURRENTLY can be faster, especially when a small number of rows are affected

Reply:
but gp need UniqueIndex? I checked `ExecRefreshMatView`.

Reply:
If there is no unique index, adding the CONCURRENTLY option poses no risk. It could improve performance if users decide to create a unique index later, so we should include this option.

Reply:
Oh, what if there is no unique index with CONCURRENTLY, I will have a check

Reply:
You'r right!
> HINT:  Create a unique index with no WHERE clause on one or more columns of the materialized view.

We should not add this options.

Reply:
Done.